optaplanner-core/src/main/java/org/optaplanner/core/impl/score/stream/drools/common/UniRuleContext.java

```diff
@@ -37,33 +39,36 @@ public UniRuleContext(Variable<A> variable, ViewItem<?>... viewItems) {
 
     public <Solution_> RuleBuilder<Solution_> newRuleBuilder(ToIntFunction<A> matchWeighter) {
         ConsequenceBuilder<Solution_> consequenceBuilder =
-                (constraint, scoreHolderGlobal) -> DSL.on(scoreHolderGlobal, variable)
-                        .execute((drools, scoreHolder, a) -> impactScore(constraint, drools, scoreHolder,
-                                matchWeighter.applyAsInt(a), a));
+                (constraint, scoreInlinerGlobal) -> DSL.on(scoreInlinerGlobal, variable)
+                        .execute((drools, scoreInliner, a) -> runConsequence(constraint, drools,
+                                scoreInliner.buildOrGetWeightedScoreImpacter(constraint),
+                                matchWeighter.applyAsInt(a),
+                                () -> singletonList(a)));
```

Comment:
Major change: scoreImpacter is now built and cached directly in the consequence.
This is necessary as we only ever know the constraint weight at rule runtime.

This is the part I don't like. It possibly slows the whole thing down.
I will have benchmark results to prove or disprove soon.

There is another possible optimization: never export the score impacter. Instead, we keep it inside the ScoreInliner, and we instead call `ScoreInliner.impactScore(Constraint, int impact)`. This way, we will be able to avoid some casting, possibly improving performance a bit. But still, it's not very nice.